ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/launcher/TracingBlock.java

```diff
@@ -61,6 +66,59 @@
 
 public class TracingBlock {
 
+	private class TracingCache {
```

Comment:
I don't think maintaining an own cache is not necessary since the TracingOptionsManager already caches the template internally and since `TracingOptionsManager.getTracingTemplate()` seem to be the expensive method (it reads the tracing options from all plugins, which is relatively heavy IO), a second cache seem unnecessary.

Reply:
I removed the cache but I left the internal class (it's now called `TracingOptionsManagerDelegate`). The reason for this is that the class provides a good encapsulation and helps me channel all calls to the "expensive methods" in `TracingOptionsManager`.